src/DotVVM.Samples.Common/ViewModels/FeatureSamples/CommandResult/SimpleExceptionFilterViewModel.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DotVVM.Framework.Hosting;
+using DotVVM.Framework.Runtime.Filters;
+using DotVVM.Framework.ViewModel;
+
+namespace DotVVM.Samples.Common.ViewModels.FeatureSamples.StaticCommandResult
+{
+    public class PageErrorModel
+    {
+        public string Message { get; set; }
+    }
+    public class ClientExceptionFilterAttribute : ExceptionFilterAttribute
+    {
+        protected override Task OnCommandExceptionAsync(IDotvvmRequestContext context, ActionInfo actionInfo, Exception exception)
+        {
+            if (exception is UIException clientError)
+            {
+                context.CommandResult = new PageErrorModel {
+                    Message = clientError.Message
+                };
```

Comment:
This changes the `CommandResult` type, which will only cause problems and should be probably forbidden.

Reply:
I agree. This is not good practice.